File: Python/Program/SivitasAkademik.py
# Import Class yang digunakan
from Human import Human
from ProgramStudi import ProgramStudi

class SivitasAkademik(Human):

    # Kelas SivitasAkademik di sini berperan sebagai:
    # - Child dari Kelas Human
    # - Parent dari Kelas Mahasiswa dan Dosen
    
    # Kelas SivitasAkademik beragregasi dengan Kelas Program Studi.
    # Hubungan Agregasi dibuat karena kelas Sivitas Akademik dan Program Studi masing-masing dapat berdiri sendiri.
    # Sebagai contoh, jika misalnya suatu objek SivitasAkademik dihapus, maka objek Program Studinya masih akan tetap ada.

    # Atribut
    __asal_universitas = ""
    __email_edu = ""
    __fakultas = ""
    __prodi = ProgramStudi()

    # Constructor
    def __init__(self, nik="", nama="", jk='',  asal_univ="", email_edu = "", fakultas="", prodi = ProgramStudi()):
        super(SivitasAkademik, self).__init__(nik, nama, jk)
        self.__asal_universitas = asal_univ
        self.__email_edu = email_edu
        self.__fakultas = fakultas
        self.__prodi = prodi

    # Setter and Getter
    def setAsalUniv(self, asal_univ):
        self.__asal_universitas = asal_univ

    def setEmailUniv(self, email_univ):
        self.__email_edu = email_univ

    def setFakultas(self, fakultas_univ):
        self.__fakultas = fakultas_univ

    def setProdi(self, prodi=ProgramStudi()):
        self.__prodi = prodi

    def getAsalUniv(self):
        return self.__asal_universitas
    
    def getEmailUniv(self):
        return self.__email_edu
    
    def getFakultas(self):
        return self.__fakultas
    
    def getProdi(self):
        return self.__prodi

File: CPP/Program/sivitasAkademik.cpp
#include "header.hh"

// == Kelas SivitasAkademik ==

// Constructors
SivitasAkademik::SivitasAkademik()
{
    this->asal_universitas = "";
    this->email_edu = "";
    this->fakultas = "";
}

SivitasAkademik::SivitasAkademik(string asal_universitas, string email_edu, string fakultas, ProgramStudi *prodi)
{
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
    this->fakultas = fakultas;
    this->prodi = prodi;
}

SivitasAkademik::SivitasAkademik(string nik, string nama, char jk, string asal_universitas, string email_edu, string fakultas, ProgramStudi *prodi) : Human(nik, nama, jk)
{
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
    this->fakultas = fakultas;
    this->prodi = prodi;
}

// Getter and Setter
string SivitasAkademik::getAsalUniv()
{
    return this->asal_universitas;
}

string SivitasAkademik::getEmailUniv()
{
    return this->email_edu;
}

string SivitasAkademik::getFakultas()
{
    return this->fakultas;
}

ProgramStudi *SivitasAkademik::getProdi()
{
    return this->prodi;
}

void SivitasAkademik::setAsalUniv(string asal_universitas)
{
    this->asal_universitas = asal_universitas;
}

void SivitasAkademik::setEmailUniv(string email_universitas)
{
    this->email_edu = email_universitas;
}

void SivitasAkademik::setFakultas(string fakultas)
{
    this->fakultas = fakultas;
}

void SivitasAkademik::setProdi(ProgramStudi *prodi)
{
    this->prodi = prodi;
}

// Destructor
SivitasAkademik::~SivitasAkademik()
{
}